products/bigquery/api.yaml

```diff
@@ -29,6 +29,8 @@ objects:
     name: 'Dataset'
     kind: 'bigquery#dataset'
     base_url: projects/{{project}}/datasets
+    self_link: projects/{{project}}/datasets/{{dataset_id}}
+    has_self_link: true
     description: |
       Datasets allow you to organize and control access to your tables.
     properties:
```

Comment:
(similar comment to elsewhere in this file) can you take a look at the description for the access block? Some of the details we have in the TF docs now would be good to keep, especially the stuff around role being required unless view is set, and one of the fields needing to be set.